core/src/main/java/greencity/controller/EventController.java

```diff
@@ -57,16 +56,12 @@ public class EventController {
     })
     @PutMapping(value = "/{eventId}", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     public ResponseEntity<EventResponseDto> update(
-            @Parameter(required = true) @EventValidation @RequestPart EventDetailsUpdate requestDto,
-            @Parameter(hidden = true) Principal principal,
-            @PathVariable Long eventId,
-            @RequestPart(required = false) @Nullable MultipartFile[] file) {
-
-        if (!eventId.equals(requestDto.getId())) {
-            throw new WrongIdException(ErrorMessage.EVENT_ID_IN_PATH_PARAM_AND_ENTITY_NOT_EQUAL);
-        }
+        @Parameter(required = true) @Valid @RequestPart EventDetailsUpdate requestDto,
+        @Parameter(hidden = true) Principal principal,
```

Comment:
You should use @CurrentUser annotation with UserVO type instead of Principal. @CurrentUser resolved by UserArgumentResolver and doesn't need Principal openly.